### PR TITLE
 Use unique symbol for wellknown symbols

### LINKED ISSUE #21603: Use unique symbol for well-known symbols
**Expected behavior:**

```ts
interface SymbolConstructor {
    readonly iterator: unique symbol;
}
```

**Actual behavior:**

```ts
interface SymbolConstructor {
    readonly iterator: symbol;
}
```

### PR DESCRIPTION
**Pull request by falsandtru**
Fixes #21603

As @rbuckton said at https://github.com/Microsoft/TypeScript/pull/15473#discussion_r149260368, we can't update `Symbol.iterator` because of @\types/node for now.

From https://github.com/Microsoft/TypeScript/commit/af73ca009369b21f2031042268ba3ccfdebdc876, unique symbol seems to have some bugs. @rbuckton Can you take a look?

### CODE REVIEW

File: tests/baselines/reference/parserSymbolProperty8.types [hunk 1]
```diff
@@ -1,6 +1,6 @@
 === tests/cases/conformance/parser/ecmascript6/Symbols/parserSymbolProperty8.ts ===
 var x: {
->x : { [Symbol.toPrimitive](): string; }
+>x : { [Symbol.toPrimitive](): string; [Symbol.toPrimitive](): string; }
```
Comment (falsandtru) [inline tests/baselines/reference/parserSymbolProperty8.types]: Duplicate.

File: tests/baselines/reference/symbolProperty18.types [hunk 2]
```diff
@@ -41,7 +41,7 @@ var str = i[Symbol.toStringTag]();
 
 i[Symbol.toPrimitive] = false;
 >i[Symbol.toPrimitive] = false : false
->i[Symbol.toPrimitive] : boolean
+>i[Symbol.toPrimitive] : any
```
Comment (falsandtru) [inline tests/baselines/reference/symbolProperty18.types]: Incorrect.

**Comment by rbuckton** [comment]
We opted not to do this yet as it will break the definitions for NodeJS (and some other packages)  in DefinitelyTyped. I'll look at this PR next week, but there may be other things we need to do first before we can address this.

**Comment by mhegazy** [comment]
We need to get lib references in first. wait for possibly a couple of releases for DT definition files to use it, and users to have compilers with support for it, then update usage in DT to use lib references, and then we can remove it from the library. 

We have a bit of time to get that done, so closing the PR in the mean while. #21603 still tracks making the switch when the time comes.

**Comment by falsandtru** [comment]
The bugs indicated here are not fixed yet. @rbuckton Can you confirm these bugs?

**Comment by mhegazy** [comment]
No, the bug is not fixed. #21603 tracks fixing it. we should also remove the special Symbol property handling in the compiler now that we have unique symbol.

**Comment by falsandtru** [comment]
I mean https://github.com/Microsoft/TypeScript/pull/21606#discussion_r165823681 and https://github.com/Microsoft/TypeScript/pull/21606#discussion_r165823702, not #21603. They should have existed in master branch.

**Comment by mhegazy** [comment]
ah.. sorry. I suppose we can take these in a separate PR.

**Comment by falsandtru** [comment]
I may make an issue several days later if nobody do it, thanks.

**Comment by falsandtru** [comment]
@rbuckton report a bug plz.

**Pull request closed without merging**